### PR TITLE
python3Packages.pyfronius: init at 0.5.2

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Python module to communicate with Fronius Symo

https://github.com/nielstron/pyfronius/

This is a Home Assistant dependency.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 128552` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.pyfronius</li>
    <li>python39Packages.pyfronius</li>
  </ul>
</details>

File: pkgs/development/python-modules/pyfronius/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    pname = "PyFronius";
+    inherit version;
+    sha256 = "077h6qh3c3dp6wdd64137m1lwp8v74k1qqi6l5cc0wgrsq8b0m1x";
+  };
+
+  propagatedBuildInputs = [
+    aiohttp
+  ];
+
+  # Project has not published tests yet
```
Comment (dotlambda) [inline pkgs/development/python-modules/pyfronius/default.nix:23]: https://github.com/nielstron/pyfronius/tree/master/pyfronius/tests

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 128552` at 16cf8003 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.pyfronius</li>
    <li>python39Packages.pyfronius</li>
  </ul>
</details>

File: pkgs/development/python-modules/pyfronius/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/pyfronius/default.nix:23]: Yes, there are tests which are not part of the source published on PyPI.

https://github.com/nielstron/pyfronius/issues/26

Comment (SuperSandro2000) [reply pkgs/development/python-modules/pyfronius/default.nix:23]: Why not fetchFromGitHub? Also does ha has tests we can enable?

Comment (dotlambda) [reply pkgs/development/python-modules/pyfronius/default.nix:23]: > Why not fetchFromGitHub?

no tags

> Also does ha has tests we can enable?

no: https://github.com/home-assistant/core/tree/2021.6.6/tests/components

Comment (dotlambda) [reply pkgs/development/python-modules/pyfronius/default.nix:23]: Maybe the comment could state that the tests are not published on PyPI and GitHub has no tags?

Comment (fabaff) [reply pkgs/development/python-modules/pyfronius/default.nix]: Upstream has added a tag yesterday. Thus, now GitHub is used and the tests are running.

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 128552` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.pyfronius</li>
    <li>python39Packages.pyfronius</li>
  </ul>
</details>

**Pull request merged**